client.go

```diff
@@ -1307,13 +1307,25 @@ func (l *link) muxDetach() {
 		Closed: true,
 		Error:  detachError,
 	}
-	select {
-	case l.session.tx <- fr:
-	case <-l.session.done:
-		if l.err == nil {
-			l.err = l.session.err
+
+Loop:
+	for {
+		select {
+		case l.session.tx <- fr:
+			// after sending the detach frame, break the read loop
+			break Loop
+		case fr := <-l.rx:
+			// discard incoming frames to avoid blocking session.mux
+			if fr, ok := fr.(*performDetach); ok && fr.Closed {
+				l.detachReceived = true
+				return
```

Comment:
This return needs to be removed so we'll go around the loop again and send the detach. Setting `l.detachReceived = true` will cause the function to return after exiting the loop.

Reply:
In this case, we would want to break the loop, but not return, right?